src/blob/core.rs

```diff
@@ -71,9 +71,11 @@ where
     }
 
     async fn write_header(&mut self) -> Result<()> {
-        let buf = serialize(&self.header)?;
-        self.file.write_append(&buf).await?;
-        self.current_offset = buf.len() as u64;
+        let mut buf = BytesMut::new();
```

Comment:
It would be good to estimate expected size and create buffer with capacity. Bincode has `serialized_size` function for that

Reply:
You can add `serialized_size` function to BLOB header struct like it is done for record header